apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs

```diff
@@ -512,6 +493,74 @@ private LogEntry BuildLogEntry(LogEventInfo loggingEvent)
             return logEntry;
         }
 
+        private static Value BuildJsonValue(object objectValue)
+        {
+            var typeCode = Convert.GetTypeCode(objectValue);
+            switch (typeCode)
+            {
+                case TypeCode.Object:
+                    if (objectValue is System.Collections.IDictionary dictionary)
+                    {
+                        var dictionaryStruct = new Struct();
+                        var itemEnumerator = dictionary.GetEnumerator();

```

Comment:
Please use a `foreach` loop over the dictionary, rather than using the enumerator directly.
If this can be done with Linq, all the better, but no problem if not.

Reply:
The default IEnumerable for IDictionary doesn't provide Key/Value-pair.

Reply:
Gosh yes, you're right. Fair enough, and thanks for the comment.